dvc/remote/local.py

```diff
@@ -342,6 +348,11 @@ def checkout(self, path_info, checksum_info, force=False):
         if bar:
             progress.finish_target(dir_relpath)
 
+    def _already_cached(self, path_info):
+        current_md5 = self.save_info(path_info).get(self.PARAM_MD5)
```

Comment:
We should use `self.state.update()` here, since checksum might be already cached in the state db.